src/controllers/idpMetadataController.ts

```diff
@@ -0,0 +1,15 @@
+import { Task } from "fp-ts/lib/Task";
+import {
+  IResponseSuccessJson,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { SuccessResponse } from "../types/commons";
+
+export default class IdpMetadataController {
+  constructor(private readonly idpMetadataRefresher: () => Task<void>) {}
+
+  public async refresh(): Promise<IResponseSuccessJson<SuccessResponse>> {
+    await this.idpMetadataRefresher().run();
```

Comment:
can you do this inside a setTimeout in order to free the socket ?

Reply:
Async upgrade added here e2acd7c
Remove the `await` could be the same

Reply:
I'm not sure it's the same (danging promise vs timeout). It depends on how exception are handled; it depends on the context but usually unhandled exception in dangling promises could cause a program crash

Reply:
probably it's ok to remove the await but yuo need to introduce a catch()

Reply:
`idpMetadataRefresher` is a `Task` that returns a Promise that never catches or it should be. Must be caught anyway?

Reply:
yes it's always a good practice to add a catch()

Reply:
Fixed in 8019d63